assets/model_monitoring/components/src/model_data_collector_preprocessor/run.py

```diff
@@ -171,7 +170,7 @@ def convert_object_to_str(dataframe: pd.DataFrame) -> pd.DataFrame:
         columns = dataframe.columns
         for column in columns:
             if dataframe[column].dtype == "object":
-                dataframe[column] = dataframe[column].astype(str)
+                dataframe[column] = dataframe[column].apply(lambda x: x if type(x) is str else json.dumps(x))
```

Comment:
should we add a UT case to verify this?

Reply:
we already have UT in test_mdc_preprocessor.test_extract_data_and_correlation_id() to cover this